packages/core-data/src/actions.js

```diff
@@ -300,44 +293,40 @@ export function* editEntityRecord( kind, name, recordId, edits, options = {} ) {
 				}, {} ),
 			},
 		},
-	};
-}
+	} );
+};
 
 /**
  * Action triggered to undo the last edit to
  * an entity record, if any.
  */
-export function* undo() {
-	const undoEdit = yield controls.select( STORE_NAME, 'getUndoEdit' );
+export const undo = () => ( { select, dispatch } ) => {
+	const undoEdit = select.getUndoEdit();
 	if ( ! undoEdit ) {
 		return;
 	}
-	yield {
+	return dispatch( {
```

Comment:
Do we need to return the result of `dispatch` here? As the action is a plain action, the return value is going to be the action object. That doesn't seem very useful. Better to return nothing, and make the `undo` return type `void`.